src/debuggability.js

```diff
@@ -141,6 +141,69 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+var supportedEvents = ["created", "chained", "fulfilled",
+    "rejected", "resolved", "cancelled"];
+
+Promise.on = function (eventName, hookFunction, simpleEventApi) {
```

Comment:
Agree. But what about performance: creating the event object and extra frame for each event will cause some overhead. Obligating all users of this API to such overhead is OK with you?